### DISCUSSION TITLE
Pin all dependencies except node. Group all node updates together.

### DISCUSSION THREAD

**Discussion by co-sic** [renovatebot/renovate#20845] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want to have all dependencies pinned and also have a group only for pinning. The exception should only be the node version. The Node version should only be fixed to the major version, e.g. 18. 

On top i want to group all the node updates into one PR. That includes:
- node version in Dockerfile (alpine image)
- node versions in github actions (actions/setup-node@v3)
- .nvmrc
- engines field in package.json

### What you tried so far.

My current (simplified) version of the renovate config:

```json
{
  "extends": ["config:base"],
  "labels": ["dependencies"],
  "node": {
    "supportPolicy": "lts"
  },
  "packageRules": [
    {
      "matchPackagePatterns": ["*"],
      "rangeStrategy": "pin"
    },
    {
      "matchDepTypes": ["engines", "peerDependencies"],
      "rangeStrategy": "auto"
    },
    {
      "matchPackageNames": ["node"],
      "rangeStrategy": "auto"
    },
    {
      "matchPaths": ["(^|/)\\.nvmrc$"],
      "versioning": "node",
      "rangeStrategy": "auto"
    },
    {
      "groupName": "nodejs",
      "matchPackageNames": ["node"],
      "schedule": ["at any time"]
    },
    {
      "groupName": "pin dependencies",
      "matchUpdateTypes": ["pin"],
      "schedule": ["at any time"]
    }
  ]
}
```

With this i manged now to get a PR for pinning all the dependencies without nodejs.

But in the pr to update nodejs (currently v16 -> v18) there are some problems:

| Package | Type | Update | Change |
|---|---|---|---|
| [node](https://togithub.com/nodejs/node) |  | major | `16` -> `18` |
| [node](https://togithub.com/nodejs/node) | engines | major | [`16.*` -> `18.*`](https://renovatebot.com/diffs/npm/node/v16.19.1/v18.15.0) |
| [node](https://togithub.com/nodejs/node) | final | major | `16-alpine` -> `19-alpine` |

It correctly updates the engines version and also nvmrc file, but the docker image is updated to v19 instead of v18. 
Additionaly it does not update the node versions from '16.x' to '18.x' in the github actions, although they are detected in the dependencyDashboard.

Any help or suggestions here would be much appreciated! Also if you see a way to make my current setup easier while archiving the same result, i'm not sure if this is the best way to do it.

**Comment by rarkins** [comment]
Pin dependencies has its own hard-coded grouping, so remove your last package rule to begin with.

It also seems like you're mixing up a few issues here, such as "how to avoid node v19 when using docker tag suffixes like -alpine" and "support for node version updating in github actions", both of which have previous discussions and maybe open issues
